### PR TITLE
feat: add support for iot drivers

### PR DESCRIPTION
**Pull request by jrk94**
#351

### CODE REVIEW

File: cmf-cli/Builders/ExecuteCommand.cs [hunk 1]
```diff
@@ -43,22 +52,37 @@ public class ExecuteCommand<T> : IBuildCommand where T : BaseCommand
...
+
         /// <summary>
         /// Executes this instance.
         /// </summary>
         /// <returns></returns>
         public Task Exec()
         {
-            dynamic command = this.Command;
-            if (this.Execute != null)
+            if (this.ConditionForExecute())
```
Comment (m-s-) [inline cmf-cli/Builders/ExecuteCommand.cs:70]: should be Condition() ?

Comment (jrk94) [reply cmf-cli/Builders/ExecuteCommand.cs:70]: you are correct

File: cmf-cli/Builders/ExecuteCommand.cs [hunk 2]
```diff
@@ -35,6 +35,14 @@ public class ExecuteCommand<T> : IBuildCommand where T : BaseCommand
         /// </value>
         public bool Test { get; set; } = false;
 
+        /// <summary>
+        /// Gets or sets the condition. This will impact the Exec
+        /// </summary>
+        /// <value>
+        /// The execute.
```
Comment (m-s-) [inline cmf-cli/Builders/ExecuteCommand.cs:42]: please review the comments :D

File: cmf-cli/Builders/ProcessCommand.cs [hunk 3]
```diff
@@ -48,41 +53,49 @@ public Task Exec()
...
-                ps.UseShellExecute = false;
-                ps.RedirectStandardOutput = true;
-                ps.RedirectStandardError = true;
 
-                if(step.EnvironmentVariables.HasAny())
+                if (Condition())
                 {
-                    foreach (var envVar in step.EnvironmentVariables)
+                    Log.Debug($"Executing '{command} {String.Join(' ', step.Args ?? Array.Empty<string>())}'");
+                    IProcessStartInfoCLI ps = ExecutionContext.ServiceProvider.GetService<IProcessStartInfoCLI>();
```
Comment (m-s-) [inline cmf-cli/Builders/ProcessCommand.cs:60]: not sure if we should remove the using statement, there are points where you can leave the execution path without disposing. maybe this should be handled in the ProcessStartInfoCLI implementation, though.

File: core/Objects/ProcessStartInfoCLI.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        {
+            this.Process.BeginErrorReadLine();
+        }
+
+        public void WaitForExit()
+        {
+            this.Process.WaitForExit();
+        }
+
+        public void Dispose()
```
Comment (m-s-) [inline core/Objects/ProcessStartInfoCLI.cs:65]: this class should probably implement IDisposable?

Comment (jrk94) [reply core/Objects/ProcessStartInfoCLI.cs:65]: it implements indirectly, through the IProcessStartInfoCLI

**Review by joaoopereira — APPROVED** [review]

**Pull request merged**